autoprof/fit/base.py

```diff
@@ -19,6 +19,29 @@ class BaseOptimizer(object):
     
     """
     def __init__(self, model, initial_state = None, relative_tolerance = 1e-3, **kwargs):
+        """
+    Initializes a new instance of the class.
+    
+    Args:
+        model (object): An object representing the model.
+        initial_state (Union[None, Tensor]): The initial state of the model. If `None`, the model's default
+                                             initial state will be used.
+        relative_tolerance (float): The relative tolerance for the optimization.
+        **kwargs (dict): Additional keyword arguments.
+        
+    Attributes:
+        model (object): An object representing the model.
+        verbose (int): The verbosity level.
+        current_state (Tensor): The current state of the model.
+        max_iter (int): The maximum number of iterations.
+        iteration (int): The current iteration number.
+        save_steps (Union[None, int]): The frequency at which to save intermediate results.
+        relative_tolerance (float): The relative tolerance for the optimization.
+        lambda_history (List[float]): A list of the optimization steps.
```

Comment:
The type here is List[ndarray], not list[float]

Reply:
Okay I'll be resolving it right now
Thanks
So I have to do it for all codes in autoprof dir?

Reply:
for now you can just change float -> ndarray in the line above. Eventually it would be great to do docstrings for every function, but that can be done in a future PR.